src/components/ModalWindowOpener.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import styled from 'styled-components';
+import { ModalWindowType } from './../model/modal-window';
+import AddUpdateMovie from './AddMovie';
+import DeleteMovie from './DeleteMovie';
+
+interface ModalWindowOpenerProps {
+  type: ModalWindowType;
+  onCloseWindow: any;
+  movieHandlers: {
+    delete: Function;
+    edit: Function;
+    add: Function;
+  };
+}
+
+export const ModalWrapper = styled.div`
+  height: 100%;
+  width: 100%;
+  position: absolute;
+  text-align: center;
+  background: black;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  z-index: 1;
+`;
+
+export const ModalContent = styled.div`
+  background: gray;
+  height: 180px;
+  width: 200px;
+  top: 200px;
+  position: relative;
+`;
+
+const ModalWindowOpener = (props: ModalWindowOpenerProps) => {
+  const onAddMovie = () => {
+    props.movieHandlers.add();
+    props.onCloseWindow();
+  };
+
+  const onUpdateMovie = () => {
+    props.movieHandlers.edit();
+    props.onCloseWindow();
+  };
+
+  const onDeleteMovie = () => {
+    props.movieHandlers.delete();
+    props.onCloseWindow();
+  };
+
+  return (
+    <ModalWrapper>
+      <ModalContent>
+        {(props.type === ModalWindowType.AddMovie && (
```

Comment:
you can do either switch case or kind of:
```tsx
const currentSection = {
  [ModalWindowType.AddMovie]: <AddUpdateMovie ... />,
  [ModalWindowType.EditMovie]: <AddUpdateMovie ... />,
}[props.type]
```

Reply:
Fixed